python-frontend/src/main/java/org/sonar/python/tree/TokenEnricher.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.tree;
+
+import com.sonar.sslr.api.Token;
+import java.util.List;
+import java.util.Map;
+import org.sonar.python.IPythonLocation;
+
+public class TokenEnricher {
+
+  private static final List<Character> ESCAPED_CHARS = List.of('"', '\'', '\\', '\b', '\f', '\n', '\r', '\t');
+
+  private TokenEnricher() {
+  }
+
+  public static List<TokenImpl> enrichTokens(List<Token> tokens, Map<Integer, IPythonLocation> offsetMap) {
+    return tokens.stream().map(token -> enrichToken(token, offsetMap)).toList();
+  }
+
+  public static TokenImpl enrichToken(Token token, Map<Integer, IPythonLocation> offsetMap) {
+    if (!offsetMap.isEmpty()) {
+      IPythonLocation location = offsetMap.get(token.getLine());
```

Comment:
While testing, I had to add 
```
if (token.getType() == PythonTokenType.DEDENT || token.getType() == GenericTokenType.EOF){
        return new TokenImpl(token);
      }
```

but it might be premature

Reply:
No, I think you are right, I should test that as we are adding these ourselves, we need to take care of these tokens. I will add some tests, and correct the behavior. And as I decided to throw rapidly this should be taken into account.

Reply:
I will create a ticket for this as it would be a good idea to enrich them. For example just not to have EOF tokens before other tokens